src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -0,0 +1,272 @@
+
+    # Trigger SAS UUT to create a PPA boundary.
+    pal_ids = [record['palId'] for record in config['palRecords']]
+    ppa_creation_request = {
+        "cbsdIds": cbsd_ids,
+        "palIds": pal_ids
+    }
+    ppa_id = self._sas_admin.TriggerPpaCreation(ppa_creation_request)
+
+    # Verify ppa_id should not be None.
+    self.assertIsNotNone(ppa_id)
+
+    # Notify the SAS UUT about the SAS Test Harness.
+    certificate_hash = getCertificateFingerprint(config['sasTestHarnessCert'])
+    self._sas_admin.InjectPeerSas({'certificateHash': certificate_hash,
+                                   'url': self._sas._base_url})
+
+    # As SAS is reset at the beginning of the test, the FAD records should contain
+    # only one zone record containing the PPA that was generated. Hence the first
+    # zone record is retrieved and verified if it matches the PPA ID.
+    response = self.TriggerFullActivityDumpAndWaitUntilComplete(
+        config['sasTestHarnessCert'], config['sasTestHarnessKey'])
+    self.assertContainsRequiredFields("FullActivityDump.schema.json", response)
+    ppa_zone_data = []
+
+    # Download 'zone' dump file.
+    for dump_file in response['files']:
+      self.assertContainsRequiredFields("ActivityDumpFile.schema.json", dump_file)
+      downloaded_file = None
+      if dump_file['recordType'] == 'zone':
+        url = str(dump_file['url']).strip()
+        downloaded_file = self._sas.DownloadFile(url, config['sasTestHarnessCert'],
+                                                 config['sasTestHarnessKey'])
+        ppa_zone_data.extend(downloaded_file['recordData'])
+
+    # Check if the retrieved FAD PPA zone record matches the PPA ID that was generated
+    # using the admin API.
+    self.assertEqual(ppa_zone_data[0]['id'], ppa_id)
+    uut_ppa_geometry = ppa_zone_data[0]['zone']['features'][0]['geometry']
+
+    # Configure the Census Tract Directory that the PPA uses
+    ppa.ConfigureCensusTractDriver(DEFAULT_CENSUSTRACTS_DATAPATH)
+
+    # PpaCreationModel requires the input registrationRequests to have 'installationParam'.
+    # But this parameter is removed for devices where conditionals are pre-loaded.
+    # Adding the 'installationParam' into registrationRequests by taking the
+    # corresponding values from conditionalRegistrationData.
+    for device in config['registrationRequests']:
+      if not device.has_key('installationParam'):
+        for conditional_params in config['conditionalRegistrationData']:
+          # Check if FCC_ID+Serial_Number present in registrationRequest
+          # and conditional_params match and add the 'installationParam'.
+          if conditional_params['fccId'] == device['fccId'] and \
+                          conditional_params['cbsdSerialNumber'] == device['cbsdSerialNumber']:
+            device.update({
+                'installationParam': conditional_params['installationParam']
+            })
+            device.update({
```

Comment:
I think you can provide some but not all REG-conditional params, so `cbsdCategory` may not exist.

Reply:
I   don’t think there is no harm to pass all REG-conditional parameter to ppa   reference model     ppa reference model would read required conditional parameters only. Anyway   to achieve precise and accuracy about what are the passing to reference   model. We have changed accordingly based on your suggestion          PPA Reference model is expecting cbsdCategory field should be present in   RegistrationRequest Object.  so we are in a position to ignore cbsdCategory

Reply:
After a second check, I think we misunderstood the comment earlier. I think that you mean some parameters are optional in REG-Conditionals and needs to be taken care. However I think that CbsdCategory is a mandatory parameter in REG-Conditionals. We will update the code accordingly. Could you please see if the below approach is good?
1.	If installParam is not present in registrationRequest then we check from the conditionals that if the device is category B.
2.	If yes then assert that all the needed parameters for PPA are present.
3.	If the device is category to value present in registrationRequest. If it is not present in registrationRequest then we set it to the value present in conditionals.
We will update this portion of code in our next commit.

Reply:
Note: Cat A can also use conditional reg data (it's not required, but they can do it). So we shouldn't assume that only Cat B devices may have conditional reg data.

I agree that if the REG-Conditional value doesn't exist in the reg request, it needs to exist in the conditional reg data. For the REG-Conditional fields outside of installationParams, they can be in the reg request and/or reg conditional data, in any combination. The value in the reg request always takes precedence. (You can see section (i) of Masoud's SAS Admins doc for more info.)

Reply:
Based   on the approach discussed we have fixed.